MpvRemote/app/src/main/java/miccah/mpvremote/MainActivity.java

```diff
@@ -167,9 +168,13 @@ public boolean onKey(View v, int keyCode, KeyEvent event) {
     void handleSendText(Intent intent) {
         String sharedText = intent.getStringExtra(Intent.EXTRA_TEXT);
         if (sharedText != null) {
-            EditText et = (EditText) findViewById(R.id.ytsearch);
-            et.setText(sharedText);
-            ytsearchEntered(sharedText);
+            // send valid url straight to server, or prepend ytdl://ytsearch for guess-searching
+            if (!URLUtil.isValidUrl(sharedText)) {
+                EditText et = (EditText) findViewById(R.id.ytsearch);
+                et.setText(sharedText);
+                sharedText = "ytdl://ytsearch:" + sharedText;
+            }
+            sendCommand(null, "play", "path", sharedText);
```

Comment:
I wonder if this logic should happen on the server side. What do you think?

Reply:
Hmm.. Good point. In that case we could unify this logic with the existing logic that checks whether it's a valid file path, which is already server-side.. I think it also makes more sense, since ytdl is very much tight to the server (mpv) rather than the client. I'll make the changes